extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchiveProcessor.java

```diff
@@ -157,6 +165,33 @@ private boolean isAdditionalBeanArchive(ApplicationArchive archive,
         return false;
     }
 
+    private boolean possiblyBeanArchive(ApplicationArchive archive) {
+        return archive.apply(tree -> {
+            boolean result = true;
+            for (String beansXml : List.of("META-INF/beans.xml", "WEB-INF/beans.xml")) {
+                result &= tree.apply(beansXml, pathVisit -> {
+                    if (pathVisit == null) {
+                        return true;
+                    }
+
+                    // crude but enough
+                    try {
+                        String text = Files.readString(pathVisit.getPath());
+                        if (text.contains("bean-discovery-mode='none'")
```

Comment:
We could also detect `all` and throw an appropriate exception in order to prevent user errors (because such archive would now still be accepted but treated as `annotated`)

Reply:
I agree that we could at least log a warning. I'm not so sure about an exception because this could break a lot of apps...

Reply:
Warning is probably a good idea.

Reply:
> because this could break a lot of apps...

Assuming they were accidentally working in the first place as we treat them as `annotated` anyway :-D
Jokes aside, I am fine with warning as well.

Reply:
Done.